### PR TITLE
mpsl: fem: increase initialization priority

### PR DESCRIPTION
**Pull request by jciupis**
Front-End Module initialization is a dependency of HCI driver initialization. Increase its priority so that there's a clear hierarchy between them.

Signed-off-by: Jędrzej Ciupis <jedrzej.ciupis@nordicsemi.no>

### CODE REVIEW

**Review by ryanjh — APPROVED** [review]

File: subsys/mpsl/fem/nrf21540_gpio/mpsl_fem_nrf21540_gpio.c [hunk 1]
```diff
@@ -293,7 +293,7 @@ static int mpsl_fem_init(const struct device *dev)
 	return fem_nrf21540_gpio_configure();
 }
 
-SYS_INIT(mpsl_fem_init, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
+SYS_INIT(mpsl_fem_init, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
```
Comment (ahasztag) [inline subsys/mpsl/fem/nrf21540_gpio/mpsl_fem_nrf21540_gpio.c:296]: I'm afraid if this is done this way at some point we will forget, why the change was introduced. Is there any way to make this initialization priority relative to SDC initialization priority?

Comment (jciupis) [reply subsys/mpsl/fem/nrf21540_gpio/mpsl_fem_nrf21540_gpio.c:296]: One way that comes to my mind is that there could be a dedicated Kconfig option representing MPSL FEM initialization priority and another one representing HCI driver initialization priority. We could then give them correct default values and put a build assert to enforce the dependency. It feels to be a bit of an overkill though. What do you think? Do you have other ideas?

Comment (ahasztag) [reply subsys/mpsl/fem/nrf21540_gpio/mpsl_fem_nrf21540_gpio.c:296]: This would be exactly the approach I think I would choose.

**Comment by rugeGerritsen** [comment]
Maybe https://github.com/nrfconnect/sdk-nrf/pull/10143 does the same thing?

**Comment by jciupis** [comment]
Superseded by #10143
